src/Part03SMContractTesting.lhs

```diff
@@ -10,15 +10,38 @@ Integration tests against state machine fakes and consumer-driven contract tests
 Motivation
 ----------
 
-- Components rarely exist in isolation, they almost always depend on some other
-  component;
-
-- When we test we often want to test as if the component existed in isolation
-  though, e.g. if component A depends on component B, we'd like to test B first
-  and then *assume* that B is working when testing A;
-
-- Assumptions like these can be justified using so called *contract tests*,
-  which is what we will be looking at next.
+So far we have seen how to test a single component sequentially ([part
+1](./docs/Part01SMTesting.md#readme)) and concurrently ([part
+2](./Part02ConcurrentSMTesting.md#readme)). Most systems are composed of several
+components however, and the global correctness of the composed system doesn't
+follow from the local correctness of its components, a typical problem being
+that the two components that are meant to talk to each other make wrong
+assumptions about each other's API.
+
+The usual solution to this problem is to add so called integration tests which
+deploy both components and perform some kind of interaction that exercises the
+API between the commonents to ensure that the assumptions are correct. Whenever
+some component needs to be deployed it will slow down the test and most likely
+introduce some flakiness related to deployment, e.g. some port is in use
+already, or not yet available to be used, or docker registry is temporarily
+down, or some other http request that is involved during deployment fails, etc.
+
+In order to avoid having slow and flaky integration tests, the standard solution
+is to mock out all the dependencies of the SUT. This works, however it
+introduces a new problem: what if the mocks are incorrect (i.e. they encode the
+same false assumptions of the consumed API). The standard solution to this
```

Comment:
Hmm, maybe not use the word `standard solution` since as the next sentence says, are actually not that common so maybe not as standard as we want.

Reply:
Fixed, thanks!